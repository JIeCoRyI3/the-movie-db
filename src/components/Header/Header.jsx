import React from 'react';
import { Route, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Logo from '../Logo';
import styles from './Header.module.css';

function Header(props) {
    console.log(props);
    return (
        <section className={styles.headerComponent}>
            <header className={styles.header}>
                <Link to="/" className={`btn btn-primary ${styles.logoHeader}`}>
                    <Logo />
                </Link>
                <Route
                    path="/film/:id"
                    render={() => (
                        <Link
                            to="/"
                            className={`btn btn-primary ${styles.goBack}`}
                        >
                            go back
                        </Link>
                    )}
                />
            </header>
            <div className={styles.headerAdditional}>{props.children}</div>
            <div className={styles.headerSortBlock}>
                <p className={styles.sortBlockTitle}>Sort by: </p>
                <button className={`btn btn-primary ${styles.sortRating}`}>
                    rating
                </button>
                <button className={`btn btn-primary ${styles.sortRelease}`}>
                    release date
                </button>
            </div>
        </section>
    );
}
Header.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.object),
        PropTypes.node,
    ]),
    match: PropTypes.object,
};

export { Header };
