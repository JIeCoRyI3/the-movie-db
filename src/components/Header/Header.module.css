.headerComponent {
    background-image: url(https://besthqwallpapers.com/Temp/14-12-2020/blue-background-with-bubble-samsung-galaxy-note-10-samsung-stock-wallpapers-blue-abstract-background-samsung-besthqwallpapers.com-1920x1080.jpg);
    border-bottom: 1px solid black;
    box-sizing: border-box;
    height: 27em;
}
.header {
    display: flex;
    justify-content: space-between;

    padding: 15px 0;
    margin-bottom: 15px;
}

.mainHeading {
    font-style: italic;
    font-size: 2.2rem;
    margin-bottom: 0;
}

.headerAdditional {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    margin-bottom: 10px;
}

.goBack {
    font-size: 1.3rem;
}

.logoHeader {
    background: none;
    outline: none;
    font-size: 2em;
    border: none;
}

.logoHeader:active,
.logoHeader:focus {
    background: none;
    outline: 0 !important;
    outline-color: transparent !important;
    outline-width: 0 !important;
    outline-style: none !important;
    box-shadow: 0 0 0 0 rgba(0, 123, 255, 0) !important;
}

.logoHeader:hover {
    background: none;
    border: none;
    outline: none;
}

.headerSortBlock {
    justify-content: flex-end;
    display: flex;
    width: 100%;
    padding-right: 5%;
    border-top: 1px solid black;
}

.sortBlockTitle {
    margin: 0.8em;
    font-size: 1.7em;
    font-style: italic;
    font-weight: bold;
}

.sortRating,
.sortRelease {
    background: none;
    border: none;
    color: black;
    font-size: 1.7em;
    font-style: italic;
    font-weight: bold;
    padding-left: 1em;
    padding-right: 0;
}
.sortRating:active,
.sortRating:focus,
.sortRelease:active,
.sortRelease:focus {
    background: none;
    color: black;
    outline: 0 !important;
    outline-color: transparent !important;
    outline-width: 0 !important;
    outline-style: none !important;
    box-shadow: 0 0 0 0 rgba(0, 123, 255, 0) !important;
}

.sortRating:hover,
.sortRelease:hover {
    background: none;
    border: none;
    outline: none;
    color: black;
}
