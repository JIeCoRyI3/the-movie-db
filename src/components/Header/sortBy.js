const sortByRating = (props) => {
    return props.byRating ? props.sortByRatingUp() : props.sortByRatingDown();
};

const sortByReleaseDate = (props) => {
    return props.byReleaseDate
        ? props.sortByReleaseDateUp()
        : props.sortByReleaseDateDown();
};

export { sortByRating, sortByReleaseDate };
