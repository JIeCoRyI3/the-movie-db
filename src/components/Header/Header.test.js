import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

const historyMock = {
    push() {
        return 1;
    },
};

describe('<Header />', () => {
    it('should renders', () => {
        const component = shallow(<Header />);
        expect(component).toMatchSnapshot();
    });

    it('should render sort by rating button', () => {
        const component = shallow(<Header />);
        expect(component.find('#rating')).toMatchSnapshot();
    });

    it('should render sort by release date button', () => {
        const component = shallow(<Header />);
        expect(component.find('#releaseDate')).toMatchSnapshot();
    });
});

describe('Sorting by rating buttons', () => {
    it('should work by rating down', () => {
        const component = shallow(<Header history={historyMock} />);
        const instance = component.instance();
        instance.props = { sortByRatingDown: null };

        const sortByRatingDownMock = jest.fn();
        instance.props.sortByRatingDown = sortByRatingDownMock;

        jest.spyOn(instance, 'sortByRating');

        component.find('#rating').simulate('click');
        expect(sortByRatingDownMock).toHaveBeenCalled();
    });

    it('should work by rating up', () => {
        const component = shallow(<Header history={historyMock} />);
        const instance = component.instance();
        instance.props = {
            sortByRatingUp: null,
            byRating: true,
        };

        const sortByRatingUpMock = jest.fn();
        instance.props.sortByRatingUp = sortByRatingUpMock;

        jest.spyOn(instance, 'sortByRating');

        component.find('#rating').simulate('click');
        expect(sortByRatingUpMock).toHaveBeenCalled();
    });
});

describe('Sorting by release date buttons', () => {
    it('should work by release date down', () => {
        const component = shallow(<Header history={historyMock} />);
        const instance = component.instance();
        instance.props = { sortByReleaseDateDown: null };

        const sortByReleaseDateDownMock = jest.fn();
        instance.props.sortByReleaseDateDown = sortByReleaseDateDownMock;

        jest.spyOn(instance, 'sortByReleaseDate');

        component.find('#releaseDate').simulate('click');
        expect(sortByReleaseDateDownMock).toHaveBeenCalled();
    });

    it('should work by release date up', () => {
        const component = shallow(<Header history={historyMock} />);
        const instance = component.instance();
        instance.props = {
            sortByReleaseDateUp: null,
            byReleaseDate: true,
        };

        const sortByReleaseDateUpMock = jest.fn();
        instance.props.sortByReleaseDateUp = sortByReleaseDateUpMock;

        jest.spyOn(instance, 'sortByReleaseDate');

        component.find('#releaseDate').simulate('click');
        expect(sortByReleaseDateUpMock).toHaveBeenCalled();
    });
});
