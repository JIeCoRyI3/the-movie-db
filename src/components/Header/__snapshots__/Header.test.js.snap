// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render button 1`] = `
<button
  className="btn btn-primary undefined"
  id="rating"
  onClick={[Function]}
>
  rating
</button>
`;

exports[`<Header /> should render sort by rating button 1`] = `
<button
  className="btn btn-primary undefined"
  id="rating"
  onClick={[Function]}
>
  rating
</button>
`;

exports[`<Header /> should render sort by release date button 1`] = `
<button
  className="btn btn-primary undefined"
  id="releaseDate"
  onClick={[Function]}
>
  release date
</button>
`;

exports[`<Header /> should renders 1`] = `
<section>
  <header>
    <Link
      className="btn btn-primary undefined"
      to="/"
    >
      <Logo />
    </Link>
    <Route
      path="/film/:id"
      render={[Function]}
    />
  </header>
  <div />
  <div>
    <p>
      Sort by: 
    </p>
    <button
      className="btn btn-primary undefined"
      id="rating"
      onClick={[Function]}
    >
      rating
    </button>
    <button
      className="btn btn-primary undefined"
      id="releaseDate"
      onClick={[Function]}
    >
      release date
    </button>
  </div>
</section>
`;
